### PR TITLE
Wire up price execution indicators

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Wiring up order execution status indicator

image

Also fixed some lint errors

# To Test

1. Unit testing!

As for manual testing, not sure how we can force if, because it depends on market price and current network fee :/

Anyway, look for the dot indicators. Should match the description (<0.5, 0.5 - 5, >5)

### CODE REVIEW

File: src/cow-react/modules/limitOrders/utils/buildPriceFromCurrencyAmounts.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+export function buildPriceFromCurrencyAmounts(
+  inputCurrencyAmount: CurrencyAmount<Currency>,
+  outputCurrencyAmount: CurrencyAmount<Currency>
+): Price<Currency, Currency>
+export function buildPriceFromCurrencyAmounts(
+  inputCurrencyAmount: CurrencyAmount<Currency> | null,
+  outputCurrencyAmount: CurrencyAmount<Currency> | null
+): Price<Currency, Currency> | null
+export function buildPriceFromCurrencyAmounts(
```
Comment (alfetopito) [inline src/cow-react/modules/limitOrders/utils/buildPriceFromCurrencyAmounts.ts:12]: For type guarding. To make TS happy, essentially.

File: src/cow-react/modules/limitOrders/utils/calculateOrderExecutionStatus.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  const referencePrice = limitPrice.greaterThan(marketPrice) ? limitPrice : marketPrice
+
+  const percentageDifference = executionPrice.divide(referencePrice).subtract(ONE)
+  const absolutePercentageDifference = percentageDifference.lessThan(ZERO)
+    ? percentageDifference.multiply(MINUS_ONE)
+    : percentageDifference
+
+  console.log(`bug-calculate`, {
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/utils/calculateOrderExecutionStatus.ts:34]: Please, use `console.debug`

File: src/cow-react/modules/limitOrders/utils/calculateOrderExecutionStatus.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    limitPrice: limitPrice.toFixed(7),
+    referencePrice: referencePrice.toFixed(7),
+    marketPrice: marketPrice.toFixed(7),
+    executionPrice: executionPrice.toFixed(7),
+    absolutePercentageDifference: absolutePercentageDifference.toFixed(7),
+  })
```
Comment (alfetopito) [inline src/cow-react/modules/limitOrders/utils/calculateOrderExecutionStatus.ts:40]: Oops, log statement left by mistake

```suggestion
```

**Review by shoom3301 — APPROVED** [review]

**Review by fairlighteth — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @alfetopito ,  I tried to test indicators, but I don't understand how to get them.
Each time when I place an order no matter if its price is below or above the market one< I see the same execution price that is a way > 5%
below and above

Some other issues I faced:
1.  when I load the app with already running limit orders there, I see different currencies in the table
different currencies

2. When I place a market order, prices I see are too weird: 
market
market1

3. I see that there is a new column on the Order history (Filled), but I don't see it on the Prorotype
no filled column

4. When I place a limit order, Executes at shows a price that is way off from the market one. However, the order is immediately executed :(
but executed
way below


Not sure that these issues are related to the current PR. Let me know please if I need to create separate issues for these points. 

Thanks!

**Comment by alfetopito** [comment]
> Hey @alfetopito , I tried to test indicators, but I don't understand how to get them. Each time when I place an order no matter if its price is below or above the market one< I see the same execution price that is a way > 5% below and above
> 
> Some other issues I faced:
> 
>     1. when I load the app with already running limit orders there, I see different currencies in the table

That's a bug not introduced in this PR.
I'll address it next, but in a different PR.

>     2. When I place a market order, prices I see are too weird:
>        market

Also not from this PR, I'll check it later maybe in a follow up if the issue continues
 
>     3. I see that there is a new column on the Order history (Filled), but I don't see it on the Prorotype

This is actually up to @fairlighteth 

>     4. When I place a limit order, Executes at shows a price that is way off from the market one. However, the order is immediately executed :(

I managed to see it when testing, I'll test again to be sure I didn't break it while refactoring

File: src/cow-react/modules/limitOrders/utils/calculateOrderExecutionStatus.ts (hunk 2 above)
Comment (alfetopito) [reply src/cow-react/modules/limitOrders/utils/calculateOrderExecutionStatus.ts]: Removed 😅

**Comment by fairlighteth** [comment]
Point 3 is long fixed on my PR. It's basically the column headers not having the same column count and not lining up correctly.

**Comment by alfetopito** [comment]
> Point 3 is long fixed on my PR. It's basically the column headers not having the same column count and not lining up correctly.

Just synced with your branch.
There's a change at every 5min, hard to keep up 😆

File: .eslintrc.json [hunk 4]
```diff
@@ -29,7 +29,7 @@
     "no-unused-vars": "off",
     "unused-imports/no-unused-imports": "error",
     "unused-imports/no-unused-vars": [
-      "warn",
+      "error",
```
Comment (anxolin) [inline .eslintrc.json:32]: 👀

File: src/cow-react/modules/limitOrders/pure/Orders/OrderRow/index.tsx [hunk 5]
```diff
@@ -122,6 +128,8 @@ export function OrderRow({
   const marketPriceInversed = isRateInversed ? prices?.marketPrice.invert() : prices?.marketPrice
   const executedPriceInversed = isRateInversed ? executedPrice?.invert() : executedPrice
 
+  const executionOrderStatus = useOrderExecutionStatus(rateInfoParams, prices)
```
Comment (anxolin) [inline src/cow-react/modules/limitOrders/pure/Orders/OrderRow/index.tsx:131]: Are we using hooks in pure components?

**Review by anxolin — COMMENTED** [review]
Great!

nice you did the test. For me was not super easy to reason about the tests. Although i really like the name you gave to the amounts. 

Not sure how they could be done more clear TBH, not sure if comments, r given/when/then with some additional comments. Anyways, happy you covered al the cases, and the definition system is kind of cool

**Pull request merged**